nixos/modules/security/pam.nix

```diff
@@ -14,6 +14,7 @@ let
       nullOr (oneOf [
         bool
         str
+        (listOf str)
```

Comment:
I assume this was added for `badwords`. We should convert `badwords` to a string instead, and reserve lists in the PAM module settings type.

The reason is that lists are a natural way to represent a key having multiple values. Without them, it would be more difficult to support a PAM module that requires the same key twice. I don't know of one off hand, but imagine a module with parameters of a form like `allow=alice allow=bob`. We would like to be able to encode that with `{ allow = [ "alice" "bob" ]; }`.